### PR TITLE
Support for google cloud vision 2 hocr by @dinosauria123

### PR DESCRIPTION
**Pull request by kba**
Works, but ideally:
- [x] use upstream repo
- [x] delete temporary files
- [ ] fall back to max x/y if width height unspecified
- [ ] maybe port to more flexible language, e.g. python

### CODE REVIEW

**Comment by zuphilip** [comment]
I am fine to start with something that works and fine-tuned that later.

_To 3:_ We cannot reconstruct the correct height, width if we only have the information from Google Cloud Vision OCR and therefore any estimation is probable semantically wrong (also it may work on a technical level). However, I suggest to drop these parameters as hard requirements and thereby also simplifying the integration here, see https://github.com/dinosauria123/gcv2hocr/issues/2.

_To 4:_ This also belongs to the upstream repo.

File: README.md [hunk 1]
```diff
@@ -163,19 +179,34 @@ capable stylesheet transformer.
...
-| From ╲ To  | hOCR               | ALTO               | PAGEXML | FineReader | Plain Text         |
-| ---:       | ---                | ---                | ---     | ---        | ---                |
-| hOCR       | -                  | :white_check_mark: | -       | -          | :white_check_mark: |
-| ALTO       | :white_check_mark: | -                  | -       | -          | :white_check_mark: |
-| PAGE       | -                  | -                  | -       | -          | -                  |
-| FineReader | -                  | -                  | -       | -          | -                  |
+| From ╲ To           | hOCR | ALTO | PAGEXML | FineReader | Plain Text  | Google Cloud Vision |
+| ---:                | ---  | ---  | ---     | ---        | ---         | ---                 |
+| hOCR                | -    | ✓    | -       | -          | ✓           | -                   |
+| ALTO                | ✓    | ✓    | -       | -          | ✓           | -                   |
```
Comment (zuphilip) [inline README.md]: The check mark for a transformation between ALTO and ALTO seems a typo. I suggest to use the equal sign `=` on the diagonal.

File: README.md [hunk 2]
```diff
@@ -163,19 +179,34 @@ capable stylesheet transformer.
...
+| PAGE                | -    | -    | -       | -          | -           | -                   |
+| FineReader          | -    | -    | -       | -          | -           | -                   |
+| Google Cloud Vision | ✓    | -    | -       | ✓          | -           | -
```
Comment (zuphilip) [inline README.md]: The second checkmark looks wrong, there is no transformation from Google Cloud Vision format to the FineReader format.

File: script/transform/gcv__hocr [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+INFILE="$1"
+OUTFILE="$2"
+#TODO
+WIDTH=2000
+HEIGHT=2000
```
Comment (zuphilip) [inline script/transform/gcv__hocr]: What is the TODO here?

**Comment by zuphilip** [comment]
We should check further the travis problems here and also the Docker image has some build errors on this branch: https://hub.docker.com/r/ubma/ocr-fileformat/builds/

**Comment by zuphilip** [comment]
I resolved the merge conflict here, this triggered new travis checks which are now passing, but the docker builds still fails:
```
make -C gcv2hocr

make[2]: Entering directory '/ocr-fileformat/vendor/gcv2hocr'
gcc -std=c99 -o gcv2hocr main.c
make[2]: Leaving directory '/ocr-fileformat/vendor/gcv2hocr'
make[1]: Leaving directory '/ocr-fileformat/vendor'

[91mmake[2]: gcc: Command not found
make[2]: *** [Makefile:9: gcv2hocr] Error 127
make[1]: *** [Makefile:112: gcv2hocr] Error 2
make: *** [Makefile:27: vendor] Error 2 
[0m
Removing intermediate container 417d39381b83
```

**Comment by kba** [comment]
Must add `gcc` to the `apk add` line and probably uninstall it afterwards since that's a pretty big package.

**Comment by zuphilip** [comment]
Okay, I tried to add this package but there is now a new error in the [docker build](https://hub.docker.com/r/ubma/ocr-fileformat/builds/be2kcdhbbx78khusp99xjgv/):
```
make -C gcv2hocr
make[2]: Entering directory '/ocr-fileformat/vendor/gcv2hocr' 
gcc -std=c99 -o gcv2hocr main.c 
[91mmain.c:1:19: fatal error: stdio.h: No such file or directory 
 #include <stdio.h> 
                   ^ 
compilation terminated.
```
Any hint how to fix this?

**Comment by zuphilip** [comment]
Okay, adding libc-dev as well did the trick...

**Comment by zuphilip** [comment]
Okay, this looks now ready to merge from my side.

**Comment by stweil** [comment]
I merged it finally. Thank you, @kba and @zuphilip.

**Comment by zuphilip** [comment]
@stweil Cool! Do you also want to do a new release `v0.2.2` now?

**Comment by stweil** [comment]
What about the remaining pull requests? Yes, it's time for a new release, but before tagging, I'd like to look after a problem report which I got for https://digi.bib.uni-mannheim.de/ocr-fileformat/ (download button not working as expected).

**Comment by zuphilip** [comment]
I don't know the status of the other PR which are just making some branches by @kba visible. Thus, these could take more time to understand and have a plan how to proceed. Fixing an error before releasing sounds certainly like a good idea.

**Pull request merged**